src/button-card.ts

```diff
@@ -0,0 +1,558 @@
+    }
+  }
+
+  private blankCardColoredHtml(state: HassEntity | undefined): TemplateResult {
+    const color = this.buildCssColorAttribute(state, undefined);
+    const fontColor = this.getFontColorBasedOnBackgroundColor(color);
+    return html`
+      <ha-card class="disabled">
+        <div class="button-card-background-color" style="color: ${fontColor}; background-color: ${color};"></div>
+      </ha-card>
+      `;
+  }
+
+  private cardColoredHtml(state: HassEntity | undefined, configState: StateConfig | undefined): TemplateResult {
+    const color = this.buildCssColorAttribute(state, configState);
+    const fontColor = this.getFontColorBasedOnBackgroundColor(color);
+    const style = this.buildStyle(state, configState);
+    return html`
+      <ha-card class="${this.isClickable(state) ? '' : "disabled"}" @ha-click="${this._handleTap}" @ha-hold="${this._handleHold}" .longpress="${longPress()}" .config="${this.config}">
+        <div class="button-card-background-color" style="color: ${fontColor}; background-color: ${color};">
+          <div class="button-card-main" style="${style}">
+            ${this.buttonContent(state, configState, "inherit")}
+          </div>
+        </div>
+        <mwc-ripple></mwc-ripple>
+      </ha-card>
+      `;
+  }
+
+  private iconColoredHtml(state: HassEntity | undefined, configState: StateConfig | undefined): TemplateResult {
+    const color = this.buildCssColorAttribute(state, configState);
+    const style = this.buildStyle(state, configState);
+    return html`
+      <ha-card class="${this.isClickable(state) ? '' : "disabled"}" @ha-click="${this._handleTap}" @ha-hold="${this._handleHold}" .longpress="${longPress()}" .config="${this.config}">
+        <div class="button-card-main" style="${style}">
+          ${this.buttonContent(state, configState, color)}
+        </div>
+        <mwc-ripple></mwc-ripple>
+      </ha-card>
+      `;
+  }
+
+  public setConfig(config: ButtonCardConfig): void {
+    if (!config) {
+      throw new Error("Invalid configuration");
+    }
+
+    this.config = {
+      tap_action: { action: "toggle" },
+      hold_action: { action: "none" },
+      size: '40%',
+      color_type: 'icon',
+      show_name: true,
+      show_state: false,
+      show_icon: true,
+      show_units: true,
+      ...config
+    };
+    this.config!.default_color = 'var(--primary-text-color)';
+    this.config!.color_off = 'var(--paper-item-icon-color)';
+    this.config!.color_on = 'var(--paper-item-icon-active-color)';
+  }
+
+  // The height of your card. Home Assistant uses this to automatically
+  // distribute all cards over the available columns.
+  public getCardSize(): number {
+    return 3;
+  }
+
+  private _handleTap(ev): void {
+    if (this.config!.confirmation &&
+      !confirm(this.config!.confirmation)) {
+      return;
+    }
+    const config = ev.target.config;
+    handleClick(this, this.hass!, config, false);
+  }
+
+  private _handleHold(ev): void {
```

Comment:
```suggestion
  private _handleHold(): void {
```
You're not using the event so don't have it as an input. Lit is smart about not passing it if you don't need it

Reply:
Oh wait, you are using it. Sorry